### ISSUE TITLE
Rule Change: multiline-comment-style option to not require newlines

### ISSUE THREAD

**Issue by RomkeVdMeulen** [eslint/eslint#17335]
### What rule do you want to change?

multiline-comment-style

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
/* A comment block
 * that aligns the starts but does not
 * begin or end with a newline. */
```

### What does the rule currently do for this code?

Under `starred-block` this comment generates two warnings:
* `Expected a linebreak after '/*'.`
* `Expected a linebreak before '*/'.`

### What will the rule do after it's changed?

Add an option for `starred-block` to disable the newline requirement. The option can default to the current behavior. E.g.:

```
/* eslint multiline-comment-style: ["error", "starred-block", { newlinesRequired: false }] */
```

### Participation

- [X] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @RomkeVdMeulen, thanks for the issue!

Per our rules policies, [stylistic rules are frozen](https://github.com/eslint/eslint#stylistic-rule-updates) so we are not adding new options to the `multiline-comment-style` rule. You can always copy the existing rule into a custom rule and make it behave the way you want.
